homeassistant/components/notion/__init__.py

```diff
@@ -205,9 +220,10 @@ def name(self) -> str:
         return f'{sensor["name"]}: {self._name}'
 
     @property
+    @async_guard_against_missing_task
```

Comment:
Since the sensor data is available inside the constructor, let's create and store the unique ID there.